packages/devtools_extensions/example/foo/packages/foo_devtools_extension/lib/src/expression_evaluation_example.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:devtools_app_shared/service.dart';
+import 'package:devtools_app_shared/ui.dart';
+import 'package:devtools_app_shared/utils.dart';
+import 'package:devtools_extensions/devtools_extensions.dart';
+import 'package:flutter/material.dart';
+
+/// A widget that shows an example of how to perform expression evaluations over
+/// the VM Service protocol.
+///
+/// The expression evaluations are made using [EvalOnDartLibrary], which is a
+/// helper class to take a library (e.g. "package:foo/src/foo_controller.dart")
+/// and perform Dart expression evaluations over it
+/// (e.g. "FooController.instance.things.value.toString()").
+/// 
+/// Evaluations can be performed when the app is both paused and unpaused. In
+/// contrast, service extension calls can only be made when the app is unpaused
+/// (see service_extension_example.dart).
+class EvalExample extends StatefulWidget {
+  const EvalExample({super.key});
+
+  @override
+  State<EvalExample> createState() => _EvalExampleState();
+}
+
+class _EvalExampleState extends State<EvalExample> with AutoDisposeMixin {
+  late final EvalOnDartLibrary fooControllerEval;
+  late final Disposable evalDisposable;
+
+  var evalResponseText = '--';
```

Comment:
Ubernit: make this a `String` and make `'--'` into a constant that can be used throughout the widget state.